cluster-autoscaler/core/scaledown/actuation/actuator_test.go

```diff
@@ -668,262 +781,294 @@ func TestStartDeletion(t *testing.T) {
 				"atomic-2-node-1": {ResultType: status.NodeDeleteErrorInternal, Err: cmpopts.AnyError},
 			},
 		},
-	} {
-		t.Run(tn, func(t *testing.T) {
-			// This is needed because the tested code starts goroutines that can technically live longer than the execution
-			// of a single test case, and the goroutines eventually access tc in fakeClient hooks below.
-			tc := tc
-			// Insert all nodes into a map to support live node updates and GETs.
-			allEmptyNodes, allDrainNodes := []*apiv1.Node{}, []*apiv1.Node{}
-			nodesByName := make(map[string]*apiv1.Node)
-			nodesLock := sync.Mutex{}
-			for _, bucket := range tc.emptyNodes {
-				allEmptyNodes = append(allEmptyNodes, bucket.Nodes...)
-				for _, node := range allEmptyNodes {
-					nodesByName[node.Name] = node
+	}
+
+	testCasesWithNGNames := map[string]startDeletionTestCase{}
+	for k, v := range testCases {
+		testCasesWithNGNames[k+" "+suffix] = v
+	}
+
+	return testCasesWithNGNames
+}
+
+func TestStartDeletion(t *testing.T) {
+	testNg1 := testprovider.NewTestNodeGroup("test", 100, 0, 3, true, false, "n1-standard-2", nil, nil)
+	opts1 := &config.NodeGroupAutoscalingOptions{
+		IgnoreDaemonSetsUtilization: false,
+	}
+	testNg1.SetOptions(opts1)
+	testNg2 := testprovider.NewTestNodeGroup("test", 100, 0, 3, true, false, "n1-standard-2", nil, nil)
+	opts2 := &config.NodeGroupAutoscalingOptions{
+		IgnoreDaemonSetsUtilization: true,
+	}
+	testNg2.SetOptions(opts2)
+
+	testSets := []map[string]startDeletionTestCase{
+		// IgnoreDaemonSetsUtilization is false
+		getStartDeletionTestCases(testNg1, opts1.IgnoreDaemonSetsUtilization, "testNg1"),
+		// IgnoreDaemonSetsUtilization is true
+		getStartDeletionTestCases(testNg2, opts2.IgnoreDaemonSetsUtilization, "testNg2"),
+	}
+
+	for _, testSet := range testSets {
+		for tn, tc := range testSet {
+			t.Run(tn, func(t *testing.T) {
```

Comment:
Code from this line and below is just a copy and paste of the older code (shifted by some lines). I haven't changed anything in the code until https://github.com/kubernetes/autoscaler/pull/5672/files#r1174841347